Backup failed when cluster is scaling

## Bug Report

### 1. Describe the bug

<!-- A clear and concise description of what the bug is. -->

Backup failed when cluster is scaling, see the following logs:

```
[2023/06/17 08:44:19.303 +08:00] [WARN] [tikv_br.go:184] ["Backup log"] [content="[2023/06/17 00:44:19.196 +00:00] [INFO] [info.go:37] [\"Welcome to Backup & Restore (BR)\"] [release-version=] [git-hash=] [git-branch=] [go-version=go1.18] [utc-build-time=\"2023-06-12 02:17:54\"] [race-enabled=false]\r\n[2023/06/17 00:44:19.197 +00:00] [INFO] [common.go:325] [arguments] [__command=\"tikv-br backup raw\"] [ca=] [cert=] [check-requirements=false] [checksum=false] [dst-api-version=V2] [end=] [format=raw] [key=] [pd=\"[http://v2tc-pd.endless-rawkv-br-tps-1808627-1-192:2379]\"] [ratelimit=64] [s3.endpoint=http://minio-peer.endless-rawkv-br-tps-1808627-1-192.svc:9000] [start=] [storage=s3://tmp/tikv-br-test/v2_br_scale/d366813c-491d-40e3-97ff-5cd29cd33b5d]\r\n[2023/06/17 00:44:19.197 +00:00] [INFO] [conn.go:244] [\"new mgr\"] [pdAddrs=v2tc-pd.endless-rawkv-br-tps-1808627-1-192:2379]\r\n[2023/06/17 00:44:19.200 +00:00] [INFO] [client.go:392] [\"[pd] create pd client with endpoints\"] [pd-address=\"[http://v2tc-pd.endless-rawkv-br-tps-1808627-1-192:2379]\"]\r\n[2023/06/17 00:44:19.204 +00:00] [INFO] [base_client.go:332] [\"[pd] update member urls\"] [old-urls=\"[http://v2tc-pd.endless-rawkv-br-tps-1808627-1-192:2379]\"] [new-urls=\"[http://v2tc-pd-0.v2tc-pd-peer.endless-rawkv-br-tps-1808627-1-192.svc:2379,http://v2tc-pd-1.v2tc-pd-peer.endless-rawkv-br-tps-1808627-1-192.svc:2379,http://v2tc-pd-2.v2tc-pd-peer.endless-rawkv-br-tps-1808627-1-192.svc:2379,http://v2tc-pd-3.v2tc-pd-peer.endless-rawkv-br-tps-1808627-1-192.svc:2379,http://v2tc-pd-4.v2tc-pd-peer.endless-rawkv-br-tps-1808627-1-192.svc:2379]\"]\r\n[2023/06/17 00:44:19.204 +00:00] [INFO] [base_client.go:350] [\"[pd] switch leader\"] [new-leader=http://v2tc-pd-4.v2tc-pd-peer.endless-rawkv-br-tps-1808627-1-192.svc:2379] [old-leader=]\r\n[2023/06/17 00:44:19.204 +00:00] [INFO] [base_client.go:105] [\"[pd] init cluster id\"] [cluster-id=7245425977568034033]\r\n[2023/06/17 00:44:19.204 +00:00] [INFO] [client.go:687] [\"[pd] tso dispatcher created\"] [dc-location=global]\r\n[2023/06/17 00:44:19.207 +00:00] [INFO] [conn.go:221] [\"checked alive KV stores\"] [aliveStores=5] [totalStores=5]\r\n[2023/06/17 00:44:19.207 +00:00] [INFO] [client.go:392] [\"[pd] create pd client with endpoints\"] [pd-address=\"[v2tc-pd.endless-rawkv-br-tps-1808627-1-192:2379]\"]\r\n[2023/06/17 00:44:19.210 +00:00] [INFO] [base_client.go:332] [\"[pd] update member urls\"] [old-urls=\"[http://v2tc-pd.endless-rawkv-br-tps-1808627-1-192:2379]\"] [new-urls=\"[http://v2tc-pd-0.v2tc-pd-peer.endless-rawkv-br-tps-1808627-1-192.svc:2379,http://v2tc-pd-1.v2tc-pd-peer.endless-rawkv-br-tps-1808627-1-192.svc:2379,http://v2tc-pd-2.v2tc-pd-peer.endless-rawkv-br-tps-1808627-1-192.svc:2379,http://v2tc-pd-3.v2tc-pd-peer.endless-rawkv-br-tps-1808627-1-192.svc:2379,http://v2tc-pd-4.v2tc-pd-peer.endless-rawkv-br-tps-1808627-1-192.svc:2379]\"]\r\n[2023/06/17 00:44:19.211 +00:00] [INFO] [base_client.go:350] [\"[pd] switch leader\"] [new-leader=http://v2tc-pd-4.v2tc-pd-peer.endless-rawkv-br-tps-1808627-1-192.svc:2379] [old-leader=]\r\n[2023/06/17 00:44:19.211 +00:00] [INFO] [base_client.go:105] [\"[pd] init cluster id\"] [cluster-id=7245425977568034033]\r\n[2023/06/17 00:44:19.211 +00:00] [INFO] [client.go:687] [\"[pd] tso dispatcher created\"] [dc-location=global]\r\n[2023/06/17 00:44:19.213 +00:00] [INFO] [client.go:80] [\"new backup client\"]\r\n[2023/06/17 00:44:19.227 +00:00] [ERROR] [client.go:847] [\"[pd] getTS error\"] [dc-location=global] [stream-addr=http://v2tc-pd-4.v2tc-pd-peer.endless-rawkv-br-tps-1808627-1-192.svc:2379] [error=\"[PD:client:ErrClientGetTSO]rpc error: code = Unknown desc = [PD:tso:ErrGenerateTimestamp]generate timestamp failed, requested pd is not leader of cluster: rpc error: code = Unknown desc = [PD:tso:ErrGenerateTimestamp]generate timestamp failed, requested pd is not leader of cluster\"] [stack=\"github.com/tikv/pd/client.(*client).handleDispatcher
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/client.go:847\"]\r\n[2023/06/17 00:44:19.228 +00:00] [INFO] [base_client.go:275] [\"[pd] cannot update member from this address\"] [address=http://v2tc-pd-0.v2tc-pd-peer.endless-rawkv-br-tps-1808627-1-192.svc:2379] [error=\"[PD:grpc:ErrGRPCDial]context canceled: context canceled\"] [errorVerbose=\"[PD:grpc:ErrGRPCDial]context canceled: context canceled
github.com/pingcap/errors.AddStack
	/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174
github.com/pingcap/errors.(*Error).GenWithStackByCause
	/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/normalize.go:307
github.com/tikv/pd/client/grpcutil.GetClientConn
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/grpcutil/grpcutil.go:56
github.com/tikv/pd/client.(*baseClient).getOrCreateGRPCConn
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/base_client.go:417
github.com/tikv/pd/client.(*baseClient).getMembers
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/base_client.go:303
github.com/tikv/pd/client.(*baseClient).updateMember
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/base_client.go:262
github.com/tikv/pd/client.(*baseClient).memberLoop
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/base_client.go:143
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1571\"]\r\n[2023/06/17 00:44:19.228 +00:00] [ERROR] [base_client.go:144] [\"[pd] failed updateMember\"] [error=\"[PD:grpc:ErrGRPCDial]context canceled: context canceled\"] [errorVerbose=\"[PD:grpc:ErrGRPCDial]context canceled: context canceled
github.com/pingcap/errors.AddStack
	/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174
github.com/pingcap/errors.(*Error).GenWithStackByCause
	/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/normalize.go:307
github.com/tikv/pd/client/grpcutil.GetClientConn
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/grpcutil/grpcutil.go:56
github.com/tikv/pd/client.(*baseClient).getOrCreateGRPCConn
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/base_client.go:417
github.com/tikv/pd/client.(*baseClient).getMembers
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/base_client.go:303
github.com/tikv/pd/client.(*baseClient).updateMember
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/base_client.go:262
github.com/tikv/pd/client.(*baseClient).memberLoop
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/base_client.go:143
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1571
github.com/tikv/pd/client.(*baseClient).updateMember
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/base_client.go:280
github.com/tikv/pd/client.(*baseClient).memberLoop
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/base_client.go:143
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1571\"] [stack=\"github.com/tikv/pd/client.(*baseClient).memberLoop
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/base_client.go:144\"]\r\n[2023/06/17 00:44:19.228 +00:00] [INFO] [base_client.go:275] [\"[pd] cannot update member from this address\"] [address=http://v2tc-pd-0.v2tc-pd-peer.endless-rawkv-br-tps-1808627-1-192.svc:2379] [error=\"[PD:grpc:ErrGRPCDial]context canceled: context canceled\"] [errorVerbose=\"[PD:grpc:ErrGRPCDial]context canceled: context canceled
github.com/pingcap/errors.AddStack
	/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174
github.com/pingcap/errors.(*Error).GenWithStackByCause
	/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/normalize.go:307
github.com/tikv/pd/client/grpcutil.GetClientConn
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/grpcutil/grpcutil.go:56
github.com/tikv/pd/client.(*baseClient).getOrCreateGRPCConn
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/base_client.go:417
github.com/tikv/pd/client.(*baseClient).getMembers
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/base_client.go:303
github.com/tikv/pd/client.(*baseClient).updateMember
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/base_client.go:262
github.com/tikv/pd/client.(*client).handleDispatcher
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/client.go:854
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1571\"]\r\n[2023/06/17 00:44:19.228 +00:00] [INFO] [client.go:706] [\"[pd] exit tso dispatcher\"] [dc-location=global]\r\n[2023/06/17 00:44:19.228 +00:00] [INFO] [collector.go:202] [\"units canceled\"] [cancel-unit=0]\r\n[2023/06/17 00:44:19.228 +00:00] [INFO] [collector.go:68] [\"Raw backup failed summary\"] [total-ranges=0] [ranges-succeed=0] [ranges-failed=0]\r\n[2023/06/17 00:44:19.228 +00:00] [ERROR] [backup.go:33] [\"failed to backup raw kv\"] [error=\"rpc error: code = Unknown desc = [PD:tso:ErrGenerateTimestamp]generate timestamp failed, requested pd is not leader of cluster\"] [errorVerbose=\"rpc error: code = Unknown desc = [PD:tso:ErrGenerateTimestamp]generate timestamp failed, requested pd is not leader of cluster
github.com/tikv/pd/client.(*client).processTSORequests
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/client.go:1081
github.com/tikv/pd/client.(*client).handleDispatcher
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/client.go:837
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1571
github.com/tikv/pd/client.(*tsoRequest).Wait
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/client.go:1297
github.com/tikv/pd/client.(*client).GetTS
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/client.go:1317
github.com/tikv/migration/br/pkg/backup.(*Client).GetTS
	/go/src/github.com/tikv/migration/br/pkg/backup/client.go:104
github.com/tikv/migration/br/pkg/backup.(*Client).UpdateBRGCSafePoint
	/go/src/github.com/tikv/migration/br/pkg/backup/client.go:142
github.com/tikv/migration/br/pkg/task.RunBackupRaw
	/go/src/github.com/tikv/migration/br/pkg/task/backup_raw.go:139
main.runBackupRawCommand
	/go/src/github.com/tikv/migration/br/cmd/br/backup.go:32
main.newRawBackupCommand.func1
	/go/src/github.com/tikv/migration/br/cmd/br/backup.go:72
github.com/spf13/cobra.(*Command).execute
	/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:852
github.com/spf13/cobra.(*Command).ExecuteC
	/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960
github.com/spf13/cobra.(*Command).Execute
	/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897
main.main
	/go/src/github.com/tikv/migration/br/cmd/br/main.go:56
runtime.main
	/usr/local/go/src/runtime/proc.go:250
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1571\"] [stack=\"main.runBackupRawCommand
	/go/src/github.com/tikv/migration/br/cmd/br/backup.go:33
main.newRawBackupCommand.func1
	/go/src/github.com/tikv/migration/br/cmd/br/backup.go:72
github.com/spf13/cobra.(*Command).execute
	/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:852
github.com/spf13/cobra.(*Command).ExecuteC
	/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960
github.com/spf13/cobra.(*Command).Execute
	/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897
main.main
	/go/src/github.com/tikv/migration/br/cmd/br/main.go:56
runtime.main
	/usr/local/go/src/runtime/proc.go:250\"]\r\n[2023/06/17 00:44:19.229 +00:00] [ERROR] [main.go:58] [\"br failed\"] [error=\"rpc error: code = Unknown desc = [PD:tso:ErrGenerateTimestamp]generate timestamp failed, requested pd is not leader of cluster\"] [errorVerbose=\"rpc error: code = Unknown desc = [PD:tso:ErrGenerateTimestamp]generate timestamp failed, requested pd is not leader of cluster
github.com/tikv/pd/client.(*client).processTSORequests
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/client.go:1081
github.com/tikv/pd/client.(*client).handleDispatcher
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/client.go:837
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1571
github.com/tikv/pd/client.(*tsoRequest).Wait
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/client.go:1297
github.com/tikv/pd/client.(*client).GetTS
	/go/pkg/mod/github.com/tikv/pd/client@v0.0.0-20220307081149-841fa61e9710/client.go:1317
github.com/tikv/migration/br/pkg/backup.(*Client).GetTS
	/go/src/github.com/tikv/migration/br/pkg/backup/client.go:104
github.com/tikv/migration/br/pkg/backup.(*Client).UpdateBRGCSafePoint
	/go/src/github.com/tikv/migration/br/pkg/backup/client.go:142
github.com/tikv/migration/br/pkg/task.RunBackupRaw
	/go/src/github.com/tikv/migration/br/pkg/task/backup_raw.go:139
main.runBackupRawCommand
	/go/src/github.com/tikv/migration/br/cmd/br/backup.go:32
main.newRawBackupCommand.func1
	/go/src/github.com/tikv/migration/br/cmd/br/backup.go:72
github.com/spf13/cobra.(*Command).execute
	/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:852
github.com/spf13/cobra.(*Command).ExecuteC
	/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960
github.com/spf13/cobra.(*Command).Execute
	/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897
main.main
	/go/src/github.com/tikv/migration/br/cmd/br/main.go:56
runtime.main
	/usr/local/go/src/runtime/proc.go:250
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1571\"] [stack=\"main.main
	/go/src/github.com/tikv/migration/br/cmd/br/main.go:58
runtime.main
	/usr/local/go/src/runtime/proc.go:250\"]\r\ncat: can't open '\r': No such file or directory\r\n"]
```

### 2. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

1. Start backup.
2. Scale (in or out) the TiKV cluster.

### 3. What did you see instead (Required)

Backup succeed.

### 4. What did you expect to see? (Required)

Backup failed.

### 5. What is your migration tool and TiKV version? (Required)

- TiKV: master (0daa38c454c8d518a62f372f5fe679c1e858871e)
- TiKV CDC: N/A
- TiKV BR: master (37f59721d954a64ba7a5e1f3a8111196a8a183ad)
- TiKV Online Bulk Load: N/A

--- a/.github/workflows/ci-br.yml
+++ b/.github/workflows/ci-br.yml
@@ -68,7 +68,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        tikv_version: [v6.1.2, nightly]
+        tikv_version: [v6.1.6, v6.5.3, v7.1.0, nightly]
         api_version: [1, 2]
         with_tls: [false, true]
         include:

--- a/.github/workflows/ci-br.yml
+++ b/.github/workflows/ci-br.yml
@@ -10,6 +10,9 @@ on:
 permissions:
   contents: read
 
+env:
+  GO_VERSION: 1.20.5
+
 jobs:
   br-check-tidy:
     runs-on: ubuntu-latest
@@ -17,7 +20,7 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-go@v2
         with:
-          go-version: '1.18.1'
+          go-version: '${{ env.GO_VERSION }}'
       - name: make check/tidy
         shell: bash
         run: |
@@ -29,7 +32,7 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-go@v2
         with:
-          go-version: '1.18.1'
+          go-version: '${{ env.GO_VERSION }}'
       - name: make check/golangci-lint
         shell: bash
         run: |
@@ -41,7 +44,7 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-go@v2
         with:
-          go-version: '1.18.1'
+          go-version: '${{ env.GO_VERSION }}'
       - name: make check/gosec
         shell: bash
         run: |
@@ -53,7 +56,7 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-go@v2
         with:
-          go-version: '1.18.1'
+          go-version: '${{ env.GO_VERSION }}'
       - name: make test_coverage
         shell: bash
         run: |
@@ -79,7 +82,7 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions/setup-go@v2
         with:
-          go-version: '1.18.1'
+          go-version: '${{ env.GO_VERSION }}'
       - name: install tiup
         run: curl --proto '=https' --tlsv1.2 -sSf https://tiup-mirrors.pingcap.com/install.sh | sh
       - name: download tikv binary
@@ -107,11 +110,15 @@ jobs:
             echo "start tikv for apiversion ${{ matrix.with_tls }} and tls enabled"
           else
             echo -e "\napi-version = ${{ matrix.api_version }}\n" >> /home/runner/work/migration/migration/.github/config/br_rawkv.toml
-            /home/runner/.tiup/bin/tiup playground ${{ matrix.tikv_version }} --mode tikv-slim --kv 1 --without-monitor --kv.config /home/runner/work/migration/migration/.github/config/br_rawkv.toml --pd.config /home/runner/work/migration/migration/.github/config/br_pd.toml &> raw.out 2>&1 &
+            /home/runner/.tiup/bin/tiup playground ${{ matrix.tikv_version }} --mode tikv-slim --host 127.0.0.1 --kv 1 --without-monitor --kv.config /home/runner/work/migration/migration/.github/config/br_rawkv.toml --pd.port 2379 --pd.config /home/runner/work/migration/migration/.github/config/br_pd.toml &> raw.out 2>&1 &
             # The first run of `tiup` has to download all components so it'll take longer.
-            sleep 1m 30s
+            timeout 180 tail -f raw.out | grep -q 'PD Endpoints'
+            if [ $? -ne 0 ]; then
+              echo "Failed to start TiKV cluster"
+              exit 1
+            fi
             # Parse PD address from `tiup` output
-            echo "PD_ADDR=$(cat raw.out | grep -oP '(?<=PD client endpoints: \[)[0-9\.:]+(?=\])')" >> $GITHUB_ENV
+            echo "PD_ADDR=127.0.0.1:2379" >> $GITHUB_ENV
             # Log the output
             echo "$(cat raw.out)" >&2
           fi

--- a/br/pkg/backup/client.go
+++ b/br/pkg/backup/client.go
@@ -101,11 +101,18 @@ func (bc *Client) GetTS(ctx context.Context, duration time.Duration, ts uint64) 
 	if ts > 0 {
 		backupTS = ts
 	} else {
-		p, l, err := bc.mgr.GetPDClient().GetTS(ctx)
+		var (
+			physical int64
+			logical  int64
+		)
+		err = utils.WithRetry(ctx, func() error {
+			physical, logical, err = bc.mgr.GetPDClient().GetTS(ctx)
+			return errors.Trace(err)
+		}, utils.NewPDReqBackoffer())
 		if err != nil {
 			return 0, errors.Trace(err)
 		}
-		backupTS = oracle.ComposeTS(p, l)
+		backupTS = oracle.ComposeTS(physical, logical)
 
 		switch {
 		case duration < 0:

--- a/br/pkg/backup/client.go
+++ b/br/pkg/backup/client.go
@@ -122,10 +122,10 @@ func (bc *Client) GetTS(ctx context.Context, duration time.Duration, ts uint64) 
 
 			backupTime := oracle.GetTimeFromTS(backupTS)
 			backupAgo := backupTime.Add(-duration)
-			if backupTS < oracle.ComposeTS(oracle.GetPhysical(backupAgo), l) {
+			if backupTS < oracle.ComposeTS(oracle.GetPhysical(backupAgo), logical) {
 				return 0, errors.Annotate(berrors.ErrInvalidArgument, "backup ts overflow please choose a smaller timeago")
 			}
-			backupTS = oracle.ComposeTS(oracle.GetPhysical(backupAgo), l)
+			backupTS = oracle.ComposeTS(oracle.GetPhysical(backupAgo), logical)
 		}
 	}
 

